Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -74,13 +78,56 @@ Ref<ViewTimeline> ViewTimeline::createFromCSSValue(Style::BuilderState& builderS
     return adoptRef(*new ViewTimeline(nullAtom(), axis, { startInset, endInset }));
 }
 
-// FIXME: compute offset values from options.
+static std::optional<Length> lengthForInset(std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>> inset)
+{
+    // TODO: Figure out how to convert CSSNumericValue to Length
+    if (holds_alternative<RefPtr<CSSNumericValue>>(inset))
+        auto value = get<RefPtr<CSSNumericValue>>(inset);
+    else if (holds_alternative<RefPtr<CSSKeywordValue>>(inset))
+        auto value = get<RefPtr<CSSKeywordValue>>(inset);
+    return { };
+}
+
+static Length lengthForInsetCSSValue(RefPtr<CSSPrimitiveValue> value)
+{
+    if (!value)
+        return { };
+    if (value->isPercentage())
+        return Length(value->resolveAsPercentageNoConversionDataRequired(), LengthType::Percent);
+    if (value->isPx())
+        return Length(value->resolveAsLengthNoConversionDataRequired<int>(), LengthType::Fixed);
+    return { };
+}
+
+static ViewTimelineInsets insetsFromOptions(std::variant<String, Vector<std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>>>> inset, CSSParserContext context)
+{
+    if (holds_alternative<String>(inset)) {
```

Comment:
You might consider using `std::get_if` here so that you can both check the type and obtain a pointer if it is the expected type. Saves the `std::get` call below.

Reply:
Fixed.